test/PackageVersionsTest/E2EInstallerTest.php

```diff
@@ -187,6 +187,60 @@ public function testRemovingPluginWithNoDevDoesNotAttemptToGenerateVersions() :
         );
     }
 
+    public function testInstallingPluginWithNoScriptsOverridesOriginalRequirements() : void
+    {
+        $this->createPackageVersionsArtifact();
+        $this->createGenericArtifact(
+            'infection-infection',
+            '0.15.0',
+            [
+                'name' => 'infection/infection',
+                'require' => [
+                    'symfony/process' => '^5.0',
+                ],
+            ]
+        );
+        $this->createGenericArtifact(
+            'symfony-process',
+            '4.0.0',
+            [
+                'name' => 'symfony/process',
+            ]
+        );
+        $this->createGenericArtifact(
+            'symfony-process',
+            '5.0.0',
+            [
+                'name' => 'symfony/process',
+            ]
+        );
+        $this->writeComposerJsonFile(
+            [
+                'name'         => 'package-versions/e2e-transitive',
+                'require'      => [
+                    'ocramius/package-versions' => '1.0.0',
+                    'infection/infection' => '^0.15.0',
```

Comment:
If I understand this correctly, this configuration will lead to `4.0.0` for the symfony version being dumped?

Reply:
No, you can check previous build where debug output was echoed - it installs 5.0 version

https://travis-ci.org/scaytrase/PackageVersions/jobs/639786230#L344-L347

But package versions detects `v4.4.3@f5697ab4cb14a5deed7473819e63141bf5352c36`

https://travis-ci.org/scaytrase/PackageVersions/jobs/642543459#L306

Reply:
Oh, 4.4.3 is not even in the set of packages you have here @_@

Reply:
Yep, that the main evidence of the problem with fallback lock - it exist there